tests/test_grid.py

```diff
@@ -766,6 +766,47 @@ def test_cast_uniform_to_rectilinear():
     assert rectilinear.bounds == grid.bounds
 
 
+def test_fft_and_rfft(noise):
+    grid = pyvista.UniformGrid(dims=(10, 10, 1))
+    with pytest.raises(ValueError, match='active point scalars'):
+        grid.fft()
+
+    full_pass = noise.fft().rfft()
+    assert full_pass['ImageScalars'].ndim == 2
+
+    # expect FFT and and RFFT to transform from time --> freq --> time domain
+    assert np.allclose(noise['scalars'], full_pass['ImageScalars'][:, 0])
+    assert np.allclose(full_pass['ImageScalars'][:, 1], 0)
+
+    output_scalars_name = 'out_scalars'
+    noise_fft = noise.fft(output_scalars_name=output_scalars_name)
+    assert output_scalars_name in noise_fft.point_data
+
+
+def test_low_pass(noise):
+    noise_no_scalars = noise.copy()
+    noise_no_scalars.clear_data()
+    with pytest.raises(ValueError, match='active point scalars'):
+        noise_no_scalars.low_pass(1, 1, 1)
+
+    with pytest.raises(ValueError, match='2 components'):
+        noise.low_pass(1, 1, 1)
+
+    out_zeros = noise.fft().low_pass(0, 0, 0)
+    assert np.allclose(out_zeros['ImageScalars'][1:], 0)
```

Comment:
I don't exactly understand what's going on here, so only asking out of caution: is `[1:]` intended here? I.e. taking every "complex" point except for the very first one? Between this and `[:, 1]` I'm getting confused.

Reply:
Fixed by using complex datatype.